### ISSUE TITLE
ls-registration-daemon posting, or not updating foreign key

### ISSUE THREAD

**Issue by soichih** [perfsonar/ls-registration-daemon#30]
Hello. My service is trying to load hosts / services records from global LS to populate data. I've seen a lot of service records that points to non-existing host record, and data lookup is failing.

For example, this service record for psmsu01.aglt2.org
> http://ps-east.es.net:8090/lookup/service/0d8241fe-ea03-4f62-85e9-7c4c3155bb20

Contains following foreign key reference
```
service-host: - [
"lookup/host/79c3203b-0e1b-43a0-8f1c-bc592b6e26fc"
],
```

When I try to load this host record, I get `ServiceRecord Not Found in DB` from the server.
> http://ps-east.es.net:8090/lookup/host/79c3203b-0e1b-43a0-8f1c-bc592b6e26fc

This host is actually stored under a different host key which is 
> http://ps-east.es.net:8090/lookup/host/af1c6929-0b0e-4e19-a151-d9e42fd98b11

I am not sure what is happening, but at the moment I am seeing quite a few services with a similar problem. Is this expected? Is there a work around?

**Comment by arlake228** [comment]
Not sure what's going on, assigning self to take a look

**Comment by soichih** [comment]
FYI.. I am seeing the same issues on following records. Not sure if it helps with troubleshooting.. but just in case.

```
Thu Jan 19 2017 14:18:38 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/b802e371-678c-4b2e-99c8-c3259018e68d statusCode:404
Thu Jan 19 2017 14:18:39 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/ea6591b7-c0a0-420c-a8f5-1ce4e1db669b statusCode:404
Thu Jan 19 2017 14:18:39 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/7530312e-17a2-4d20-af25-8bff7820487e statusCode:404
Thu Jan 19 2017 14:18:39 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/f1ba1b9b-a64d-4350-8cad-aa3dd081b975 statusCode:404
Thu Jan 19 2017 14:18:39 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/c640dd3a-8dc3-4713-ae12-8e91d9f0f13f statusCode:404
Thu Jan 19 2017 14:18:41 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/0db87f69-96e1-46a6-9d50-efc3df20270d statusCode:404
Thu Jan 19 2017 14:18:41 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/325360ab-616d-4488-8410-88d9d5a5662b statusCode:404
Thu Jan 19 2017 14:18:42 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/75383a4c-55a6-49b0-a681-765348a9c8a6 statusCode:404
Thu Jan 19 2017 14:18:42 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/9d9ccacb-d860-4848-a43d-d9f16bca28c1 statusCode:404
Thu Jan 19 2017 14:18:43 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/53d18d0c-3f0b-4af3-92e2-591cebfc9df2 statusCode:404
Thu Jan 19 2017 14:18:43 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/adecba1f-7a77-4c95-9401-92c0c2b47683 statusCode:404
Thu Jan 19 2017 14:18:43 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/ff27f57d-f4d3-457d-911d-d61e73f735cf statusCode:404
Thu Jan 19 2017 14:18:44 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/25f0607a-7d8e-4285-9928-70d69cb434b5 statusCode:404
Thu Jan 19 2017 14:18:44 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://monipe-ls.rnp.br:8090/lookup/host/a8a5e24d-30a9-46f2-bc60-86d8b60aad67 statusCode:404
Thu Jan 19 2017 14:18:45 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://monipe-ls.rnp.br:8090/lookup/host/f40addbe-ae40-46b6-a4cb-30b902047f33 statusCode:404
Thu Jan 19 2017 14:18:47 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-west.es.net:8090/lookup/host/28ad155c-2b5b-446c-a27a-645fb33c5205 statusCode:404
Thu Jan 19 2017 14:18:49 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/35aede92-90b4-4e30-b00e-c5440b178d4a statusCode:404
Thu Jan 19 2017 14:18:49 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/528331f1-ebf7-4645-a53c-97b025d43586 statusCode:404
Thu Jan 19 2017 14:18:50 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/12c7d5f6-0181-4628-bc59-45cc7a75ffb4 statusCode:404
Thu Jan 19 2017 14:18:50 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/76686c45-ba2c-4c01-92eb-38ffe6b54435 statusCode:404
Thu Jan 19 2017 14:18:51 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://monipe-ls.rnp.br:8090/lookup/host/2aebb179-e726-4762-99fa-3531afa86143 statusCode:404
Thu Jan 19 2017 14:18:52 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/4fa0ecdb-19c4-4920-a6e7-b29d491296d6 statusCode:404
```

**Comment by soichih** [comment]
@arlake228 I started seeing this issue again. 

http://nsw-brwy-sls1.aarnet.net.au:8090/lookup/service/3cbf6ac2-ca7b-4356-8886-ecdab3fc2eb5
http://nsw-brwy-sls1.aarnet.net.au:8090/lookup/host/ac80f86a-18ff-40b0-8286-25137d6c2796

In case the issue disappears..

![s2](https://cloud.githubusercontent.com/assets/923896/24307362/466c7198-109a-11e7-894e-e498f9651b91.png)
![s1](https://cloud.githubusercontent.com/assets/923896/24307363/466cf316-109a-11e7-8e11-18563ebbe6dd.png)

```
21|mccache | Fri Mar 24 2017 17:49:22 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-west.es.net:8090/lookup/host/f3108176-5ab8-4dbf-bc8c-e9515be39c5a statusCode:404
21|mccache | Fri Mar 24 2017 17:49:26 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:26 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/592d301a-a581-476c-a1f9-701185bd3c10 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:26 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:26 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/a95b2887-4f0f-4b6c-b3bd-a5564f9eb37f statusCode:404
21|mccache | Fri Mar 24 2017 17:49:26 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:26 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/2b16dffb-c9ca-4d5d-a089-465a3f306249 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:26 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:26 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/91cb6fc3-7457-40ef-892b-aabf599aaad3 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:27 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:27 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/b9a76e39-ba1f-40ae-80f1-d1e2c3415282 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:27 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:27 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/c903c493-8b4f-410a-8f63-ad346ddcc06c statusCode:404
21|mccache | Fri Mar 24 2017 17:49:28 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:28 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/40ea814f-9712-4c52-8aef-85a6f7da3c37 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:29 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:29 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/c8fecf82-6ce2-44a8-a2e3-9929e51b5c5f statusCode:404
21|mccache | Fri Mar 24 2017 17:49:30 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:30 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/437ef2e7-1d73-4fb8-ae8b-38d92b6b346d statusCode:404
21|mccache | Fri Mar 24 2017 17:49:30 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:30 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/da84df12-7c52-4d41-a33d-3d7a4597e84a statusCode:404
21|mccache | Fri Mar 24 2017 17:49:30 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:30 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/76b692e0-b703-45be-805b-075f0a5d0d4d statusCode:404
21|mccache | Fri Mar 24 2017 17:49:32 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:32 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://monipe-ls.rnp.br:8090/lookup/host/e0909464-92ee-41f6-bb6c-3284131dc8e9 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:33 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:33 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://nsw-brwy-sls1.aarnet.net.au:8090/lookup/host/ac80f86a-18ff-40b0-8286-25137d6c2796 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:34 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:34 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-west.es.net:8090/lookup/host/7f8c9cdb-51e1-4029-847f-af1607532277 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:36 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:36 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/68c3e20a-cfe3-45b1-87a0-2684442ef441 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:36 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:36 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/4813b448-f3c3-4bd3-b786-18b9405104d3 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:36 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:36 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/6d1dadce-0b89-4cf7-906b-0792a788b371 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:37 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:37 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/20af360f-2ec4-4728-8248-3db0633cd4d2 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:38 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:38 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://monipe-ls.rnp.br:8090/lookup/host/a1834e89-9c29-49e0-85f1-36537e928853 statusCode:404
21|mccache | Fri Mar 24 2017 17:49:39 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:39 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-west.es.net:8090/lookup/host/b4c95c40-2c39-4cce-bbed-5e49f580148f statusCode:404
21|mccache | Fri Mar 24 2017 17:49:40 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:40 GMT+0000 (UTC) - error:  Error: failed to cache host from:http://ps-east.es.net:8090/lookup/host/c1501d70-e279-4fc9-bcd5-a5c2a551935f statusCode:404
21|mccache | Fri Mar 24 2017 17:49:51 GMT+0000 (UTC) - error: failed to create hostrecord from following service
21|mccache | Fri Mar 24 2017 17:49:54 GMT+0000 (UTC) - error: failed to create hostrecord from following service

```

**Comment by arlake228** [comment]
Looked at this, the fix is in the 4.0 tree which just had RC3 go out today, which means most people do not have it. We should be doing 4.0 final in a few weeks which should greatly increase the number of people that have this fix. As I said in chat, its client-side unfortunately so I can't guarantee 100% coverage since not everyone will upgrade right away, but it will get better.
